anaconda-modules.ks.in

```diff
@@ -0,0 +1,48 @@
+#version=DEVEL
+#test name: anaconda-modules
+
+# Verify that only the enabled Anaconda DBus modules are started.
+
+# Use defaults.
+%ksappend repos/default.ks
+%ksappend common/common_no_payload.ks
+%ksappend payload/default_packages.ks
+
+%post --nochroot
+
+# Get the activated Anaconda DBus modules.
+less /tmp/syslog | grep "Activating service name" | \
+grep "org.fedoraproject.Anaconda.Modules" | cut -d"'" -f2 \
+| sort > /tmp/generated.out
+
+# Generate the expected output.
+cat > /tmp/expected.out <<EOF
+org.fedoraproject.Anaconda.Modules.Localization
+org.fedoraproject.Anaconda.Modules.Network
+org.fedoraproject.Anaconda.Modules.Payloads
+org.fedoraproject.Anaconda.Modules.Security
+org.fedoraproject.Anaconda.Modules.Services
+org.fedoraproject.Anaconda.Modules.Storage
+org.fedoraproject.Anaconda.Modules.Subscription
+org.fedoraproject.Anaconda.Modules.Timezone
+org.fedoraproject.Anaconda.Modules.Users
+EOF
+
+# Drop the Subscription module if this is not RHEL.
+if [[ "@KSTEST_OS_NAME@" != "rhel" ]]; then
+    sed -i '/Subscription/d' /tmp/expected.out
+fi
+
+# Check the output
+diff /tmp/expected.out /tmp/generated.out
+
+if [[ $? != 0 ]]; then
+    echo "*** unexpected modules were activated:" >> /mnt/sysroot/root/RESULT
+    cat /tmp/generated.out >> /mnt/sysroot/root/RESULT
+    echo "*** the following modules were expected:" >> /mnt/sysroot/root/RESULT
+    cat /tmp/expected.out >> /mnt/sysroot/root/RESULT
```

Comment:
And this is pure speculation, but I'm also wondering about using output of `diff -u 20 ...`, that could be easier to see and compare at once, unless the lists go totally haywire or grow to tens of items.

Reply:
There is a huge space for improvements, but I don't really want to play with it too much.